src/TrieMap.mo

```diff
@@ -66,23 +142,80 @@ module {
       ov
     };
 
-    /// An `Iter` over the keys.
+    /// Returns an iterator over the keys of the map.
+    ///
+    /// Each iterator gets a _persistent view_ of the mapping, and is unaffected
+    /// by concurrent updates to the iterated map.
+    ///
+    /// Example:
+    /// ```motoko include=initialize
+    /// map.put(0, 10);
+    /// map.put(1, 11);
+    /// map.put(2, 12);
+    ///
+    /// // find the sum of all the keys
+    /// var sum = 0;
+    /// for (key in map.keys()) {
+    ///   sum += key;
+    /// };
+    /// // 0 + 1 + 2
+    /// sum
+    /// ```
     ///
-    /// Each iterator gets a _persistent view_ of the mapping, independent of concurrent updates to the iterated map.
+    /// Runtime: O(1)
+    /// Space: O(1)
     public func keys() : I.Iter<K> {
```

Comment:
Just an idea, not important: Should we mention that iteration takes linear time? Only returning the iterator object is O(1). Space is maybe also not O(1) during iteration because of temporary object created during traversal?

Reply:
Hahaha yes I was being a bit cheeky with this analysis. I will add an extra note about the actual iteration.